src/lib/operators/aggregate_hash.cpp

```diff
@@ -946,6 +952,8 @@ void AggregateHash::write_aggregate_output(ColumnID aggregate_index) {
         std::make_shared<ValueSegment<decltype(aggregate_type)>>(std::move(values), std::move(null_values));
   }
   _output_segments[output_column_id] = output_segment;
+
+  aggregate_columns_writing_duration += timer.lap() - write_groupby_output_duration;
```

Comment:
`_write_groupby_output` is only called for `aggregate_index == 0`. Did you test this?

Reply:
Not sure what you are asking for. `write_groupby_output_duration` does only store non-zero runtimes, when  `aggregate_index == 0` was true. Was this the question?

Reply:
I confused it with `groupby_columns_writing_duration`. Maybe, we can find clearer names? Also, should we a comment explaining that `write_groupby_output_duration` is 0 for all but the first aggregate?

Reply:
Like this?

```
// Used to track the duration of groupby columns writing, which is done for the first aggregate only. Value is
  // substraced from the runtime of this method (thus, it's either 0 or non-zero for the first aggregate).
  auto first_aggregate_groupby_column_writing_time = std::chrono::nanoseconds{};
```

Don't like the var name.

Reply:
Typo: subtracted

Maybe just call it `excluded_time`?

Reply:
Changed.